src/library/trackcollection.cpp

```diff
@@ -510,4 +516,68 @@ StorageUnicodeFunctions::likeFunction(sqlite3_context *p,
                                     itrString, endString, E));
 }
 */
+//Private function
+
+bool TrackCollection::settingsPathStatus(){
+    struct stat sb; //this struct will hold the access info of the dir using struct here as below it has function of same name
+    stat(m_pConfig->getSettingsPath().toStdString().c_str(), &sb);  //give the status, m_pConfig->getSettingsPath() will give the
+                                                                   //directory in QString which is conver. to C style string
+    int n = sb.st_mode; //This is the file access number and in octal form represents the file access its of form 40777
+    n = n % 512; //As we need last 3 digits in the octal form so moding it with 8^3
+    //Now in octal form, first digit will represent the owner permission and we only need it so 
+    n = n / 64;
+    switch(n)
+    {
+        case 0:
```

Comment:
We should avoid literal constants in our code. In this case you could use the macros form 
stat.h

But as said, switch to Qt functions.